src/vm.rs

```diff
@@ -352,6 +357,27 @@ impl<T: GuestStagePagingMode> Vm<T, VmStateInitializing> {
 
     /// Completes intialization of the `Vm`, returning it in a finalized state.
     pub fn finalize(self) -> Result<Vm<T, VmStateFinalized>> {
+        // If AIA virtualization is enabled, make sure all vCPUs got assigned an IMSIC address and
+        // that there are no aliases.
+        for i in 0..self.vcpus.num_vcpus() {
```

Comment:
nit: this might be easier to read and test if contained in a function like `has_imsic_overlap` or similar.

Reply:
Done, moved this to a separate helper method.